src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.Statements.cs

```diff
@@ -152,6 +152,28 @@ private void EmitLoopExpression(Expression expr)
         private void EmitSwitchExpression(Expression expr, CompilationFlags flags)
         {
             SwitchExpression node = (SwitchExpression)expr;
+            
+            if (node.Cases.Count == 0)
+            {
```

Comment:
I looked through both compiler and interpreter and it appears that they consider only particular ExpressionTypes as reducible, so they will not attempt to reduce the switch.

Also in both cases, once the node is one of the reducible ExpressionTypes, it is treated as unconditionally reducible - for example interpreter throws if a node of reducible ExpressionType returns false from CanReduce and stack spiller in the compiler will force rewriting of all the parent nodes regardless of what CanReduce reports.

It would not be easy to treat SwitchExpression the same as other reducible expressions since it is "sometimes reducible".